chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,222 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
+    pubkey = args.first().atom
+    args = args.rest()
+    message = args.first().atom
+    if len(pubkey) != 48:
+        raise ValidationError(Err.INVALID_CONDITION)
+    if len(message) > 1024:
+        raise ValidationError(Err.INVALID_CONDITION)
+    return [pubkey, message]
+
+
+def parse_create_coin(args: SExp) -> List[bytes]:
+    puzzle_hash = args.first().atom
+    args = args.rest()
+    amount = args.first().atom
+    if len(puzzle_hash) != 32:
+        raise ValidationError(Err.INVALID_CONDITION)
+    amount_int = int_from_bytes(amount)
+    if amount_int >= 2 ** 64:
+        raise ValidationError(Err.COIN_AMOUNT_EXCEEDS_MAXIMUM)
+    if amount_int < 0:
+        raise ValidationError(Err.COIN_AMOUNT_NEGATIVE)
+    # note that this may change the representation of amount. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [puzzle_hash, int_to_bytes(amount_int)]
+
+
+def parse_seconds(args: SExp, error_code: Err) -> Optional[List[bytes]]:
+    seconds = args.first().atom
+    seconds_int = int_from_bytes(seconds)
+    # this condition is inherently satisified, there is no need to keep it
+    if seconds_int <= 0:
+        return None
+    if seconds_int >= 2 ** 64:
+        raise ValidationError(error_code)
+    # note that this may change the representation of seconds. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(seconds_int)]
+
+
+def parse_height(args: SExp, error_code: Err) -> Optional[List[bytes]]:
+    height = args.first().atom
+    height_int = int_from_bytes(height)
+    # this condition is inherently satisified, there is no need to keep it
+    if height_int <= 0:
+        return None
+    if height_int >= 2 ** 32:
+        raise ValidationError(error_code)
+    # note that this may change the representation of the height. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(height_int)]
+
+
+def parse_fee(args: SExp) -> List[bytes]:
+    fee = args.first().atom
+    fee_int = int_from_bytes(fee)
+    if fee_int >= 2 ** 64 or fee_int < 0:
+        raise ValidationError(Err.RESERVE_FEE_CONDITION_FAILED)
+    # note that this may change the representation of the fee. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(fee_int)]
+
+
+def parse_coin_id(args: SExp, error_code: Err) -> List[bytes]:
+    coin = args.first().atom
+    if len(coin) != 32:
+        raise ValidationError(error_code)
+    return [coin]
+
+
+def parse_hash(args: SExp, error_code: Err) -> List[bytes]:
+    h = args.first().atom
+    if len(h) != 32:
+        raise ValidationError(error_code)
+    return [h]
+
+
+def parse_amount(args: SExp) -> List[bytes]:
+    amount = args.first().atom
+    amount_int = int_from_bytes(amount)
+    if amount_int < 0:
+        raise ValidationError(Err.ASSERT_MY_AMOUNT_FAILED)
+    if amount_int >= 2 ** 64:
```

Comment:
I was considering using the `MAX_COIN_AMOUNT`, but it's a bit messy to thread that constant all the way down here. Additionally, in a fair number of places, we assign coin amounts to `uint64` integers, without caring about the `MAX_COIN_AMOUNT` constant. So it seems safe to expect 2**64 to be one past the largest amount allowed for coins (and fees, this comment applies to `parse_fee()` as well)